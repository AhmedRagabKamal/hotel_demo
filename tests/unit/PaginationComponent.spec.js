import { mount } from '@vue/test-utils';
import Pagination from '@/shared/components/PaginationComponent.vue';
// import sinon from 'sinon';

describe('Pagination', () => {
  const build = () => {
    const wrapper = mount(Pagination, {
      propsData: {
        perPage: 3,
        totalRows: 20,
        reviews: [],
      },
    });
    return {
      wrapper,
    };
  };

  it('renders the Pagination component', () => {
    // arrange
    const { wrapper } = build();
    // assert
    expect(wrapper.html()).toMatchSnapshot();
    expect(wrapper.html()).toContain('<nav aria-label="Page navigation example">');
  });

  it('has a button', () => {
    const { wrapper } = build();
    expect(wrapper.contains('li.page-item')).toBe(true);
  });

  it('pagination button click should increment the page number', () => {
    const { wrapper } = build();
    expect(wrapper.vm.pageNumber).toBe(0);
    const button = wrapper.find('button.next');
    button.trigger('click');
    expect(wrapper.vm.pageNumber).toBe(1);
  });
});
