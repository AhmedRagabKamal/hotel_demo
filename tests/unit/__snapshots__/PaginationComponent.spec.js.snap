// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination renders the Pagination component 1`] = `
<nav aria-label="Page navigation example">
  <ul class="pagination justify-content-end">
    <li class="page-item"><button aria-label="Previous" disabled="disabled" class="page-link"><span aria-hidden="true">«</span></button></li>
    <li class="page-item active"><button class="page-link">1</button></li>
    <li class="page-item"><button class="page-link">2</button></li>
    <li class="page-item"><button class="page-link">3</button></li>
    <li class="page-item"><button class="page-link">4</button></li>
    <li class="page-item"><button class="page-link">5</button></li>
    <li class="page-item"><button class="page-link">6</button></li>
    <li class="page-item"><button class="page-link">7</button></li>
    <li class="page-item"><button aria-label="Next" class="page-link next"><span aria-hidden="true">»</span></button></li>
  </ul>
</nav>
`;
